code/datums/diseases/advance/symptoms/vomit.dm

```diff
@@ -53,11 +53,21 @@ and your disease can spread via people walking on vomit.
 		else
 			vomit(M)
 
-/datum/symptom/vomit/proc/vomit(mob/living/carbon/M)
+/datum/symptom/vomit/proc/vomit(mob/living/carbon/vomiter)
+	var/deductable_nutrition = 0
+	var/constructed_flags = (MOB_VOMIT_MESSAGE | MOB_VOMIT_HARM)
+	var/type_of_vomit = /obj/effect/decal/cleanable/vomit/toxic
 	if(vomit_nebula)
-		M.vomit(lost_nutrition = 10, blood = vomit_blood, vomit_type = VOMIT_NEBULA, stun = FALSE, distance = proj_vomit)
+		type_of_vomit = /obj/effect/decal/cleanable/vomit/nebula
+		deductable_nutrition = 10
 	else
-		M.vomit(lost_nutrition = 20, blood = vomit_blood, distance = proj_vomit)
+		constructed_flags |= MOB_VOMIT_STUN
+		deductable_nutrition = 20
```

Comment:
ok we're cooking now one of my only remaining concerns is that this is still bad OOP to me
we could just have `type_of_vomit` and `deductable_nutrition` as parameters of this type instead of this either/or check...
if i dont find anything else while scrolling i dont care a huge amount though

Reply:
parameters of the type? what? this is a totally different `vomit()` proc on the `/datum/symptom` btw, we have to translate the extant symptoms framework into what we can invoke on carbons. i don't think this is the time to shuffle everything around in response to this though